lib/elixir/lib/map.ex

```diff
@@ -384,13 +384,20 @@ defmodule Map do
       %{a: 1, c: 3}
 
   """
-  @spec take(map, Enumerable.t()) :: map
+  @spec take(map, [key]) :: map
   def take(map, keys)
```

Comment:
```suggestion
```
the head definition can be removed here. and same with the other functions.

Is it desirable to have that in the Elixir code.
It reads better. I'm wondering if new code should be like that.

Reply:
I disagree on this. I think when you have multiple clauses with different argument names, it's definitely better to have a bodiless clause. This allows you to see the function signature at a glance and ensures the correct name of the arguments in the documentation, instead of relying on the first clause. Many times I've seen the first clause get changed and screw up with the arguments in the docs because one forgets about updating that.

Reply:
Sorry, I expressed it wrong. I was saying it read better with the bodiless clause, but i think we have been using mostly without thia clause. Should we start relying on it?

Reply:
It's a case by case situation. I don't want to make it a "rule" in any way. If there are a few clauses and the argument names change between the clauses, I tend to go with a bodiless clause.